app/code/Magento/MediaGallery/Test/Unit/Model/Asset/Command/GetByIdTest.php

```diff
@@ -0,0 +1,119 @@
+
+/**
+ * Test the GetById command model
+ */
+class GetByIdTest extends TestCase
+{
+    private const MEDIA_ASSET_STUB_ID = 1;
+
+    private const MEDIA_ASSET_DATA = ['id' => 1];
+
+    /**
+     * @var GetById|MockObject
+     */
+    private $getMediaAssetById;
+
+    /**
+     * @var AssetInterfaceFactory|MockObject
+     */
+    private $assetFactory;
+
+    /**
+     * @var AdapterInterface|MockObject
+     */
+    private $adapter;
+
+    /**
+     * @var Select|MockObject
+     */
+    private $selectStub;
+
+    /**
+     * @var Statement|MockObject
+     */
+    private $statementMock;
+
+    /**
+     * @var LoggerInterface|MockObject
+     */
+    private $logger;
+
+    /**
+     * Initialize basic test class mocks
+     */
+    protected function setUp(): void
+    {
+        $resourceConnection = $this->createMock(ResourceConnection::class);
+        $this->assetFactory = $this->createMock(AssetInterfaceFactory::class);
+        $this->logger = $this->createMock(LoggerInterface::class);
+
+        $this->getMediaAssetById = (new ObjectManager($this))->getObject(
+            GetById::class,
+            [
+                'resourceConnection' => $resourceConnection,
+                'assetFactory' => $this->assetFactory,
+                'logger' =>  $this->logger,
+            ]
+        );
+        $this->adapter = $this->createMock(AdapterInterface::class);
+        $resourceConnection->method('getConnection')->willReturn($this->adapter);
+
+        $this->selectStub = $this->createMock(Select::class);
+        $this->selectStub->method('from')->willReturnSelf();
+        $this->selectStub->method('where')->willReturnSelf();
+        $this->adapter->method('select')->willReturn($this->selectStub);
+
+        $this->statementMock = $this->getMockBuilder(\Zend_Db_Statement_Interface::class)->getMock();
+    }
+
+    /**
+     * Test successful get media asset by id command execution.
+     */
+    public function testSuccessfulGetByIdExecution(): void
+    {
+        $this->statementMock->method('fetch')->willReturn(self::MEDIA_ASSET_DATA);
+        $this->adapter->method('query')->willReturn($this->statementMock);
+
+        $mediaAssetStub = $this->getMockBuilder(AssetInterface::class)->getMock();
+        $this->assetFactory->expects($this->once())->method('create')->willReturn($mediaAssetStub);
+
+        $this->getMediaAssetById->execute(self::MEDIA_ASSET_STUB_ID);
+    }
+
+    /**
+     * Test case when there is no found media asset by id.
+     */
+    public function testNotFoundMediaAssetException(): void
+    {
+        $this->statementMock->method('fetch')->willReturn([]);
+        $this->adapter->method('query')->willReturn($this->statementMock);
+
+        $this->expectException(IntegrationException::class);
+        $this->logger->expects($this->once())
+            ->method('critical')
+            ->willReturnSelf();
+
+        $this->getMediaAssetById->execute(self::MEDIA_ASSET_STUB_ID);
```

Comment:
Please, add an assertion of the returned result here as well.

Reply:
Hello @rogyar , thank you for your review. I have a question about this: do we need to have an assertion here when we already expect an exception: ` $this->expectException(IntegrationException::class);` In that case, I think that assertion is satisfied by the fact of throwing an integration exception. What assertion you would like to see here?

Reply:
I don't think the assertion is necessary here

Reply:
Agree, missed the point that the test already asserts the exception. Thank you